docs/development/apifirst.md

```diff
@@ -0,0 +1,118 @@
+
+### Good patterns for building API First Fullstack Node.JS solutions
+
+1. Prefer generating code from OpenAPI file for both client and server. Generating code based on the specification will ensure that the same types, formats are used. This will enable your team to iterate on the specification without worry of getting out of sync.
+
+2. When making changes in the OpenAPI file change it's [version](https://github.com/OAI/OpenAPI-Specification/blob/main/examples/v3.0/petstore-expanded.yaml#L3). Changed version will help others to detect what kind of changes were made.
+
+3. When introducing breaking changes consider adding them as new endpoints by introducing another v2 prefix in the API path. 
+ 
+4. Every path should have `operationId` field specified. This field is used by generator to generate method names for clients etc.
+
+5. When building response objects preffer referencing predefined Objects and Enums in [Schemas](https://swagger.io/docs/specification/data-models/)
+
+6. If API return specific [error object](https://github.com/OAI/OpenAPI-Specification/blob/main/examples/v3.0/petstore-expanded.yaml#L148-L158) it should be defined in Schemas.
+
+7. Declare servers and security scheme to enable users to call API from OpenAPI Editor and other tools. 
+
+8. Use [tags](https://swagger.io/docs/specification/grouping-operations-with-tags/) to define namespaces for your API. Grouping operations with tags that will be used to organize your generated source code into folder structure.
+
+// TODO move this to the top
+
+## Recommended Packages
+
+List bellow provides comprehensive set of libraries that can be used for end to end full stack application written in Node.JS, Express.js and client side application.
+
+### Code Generation Tools
+
+#### @openapitools/openapi-generator-cli - <https://openapi-generator.tech>
+
+CLI provides support for generating source code based on the OpenAPI spec. CLI supports wide range of languages and frameworks:
+
+https://openapi-generator.tech/docs/generators
+
+CLI has widespread usage across industry including many community projects at Red Hat.
+
+Project is maintained by OpenAPI Generator Community
+
+### Node.js backend generator
+
+#### nodejs-express-server - <https://openapi-generator.tech/docs/generators/nodejs-express-server>
+
+This generator generates Express.js based stub implementation based on your OpenAPI file. 
+Generation can be done by using openapi-generator cli:
+
+```bash
+openapi-generator generate -g nodejs-express-server -i yourapi.json -o ./yourproject
+```
+
+### Client Side generator
+
+#### typescript-node - <https://openapi-generator.tech/docs/generators/typescript-node>
+
+This generator generates client for Node.js backend that allows us to perform requests against another backend
+against a
+
+```bash
+openapi-generator generate -g typescript-node -i yourapi.json -o ./yourproject
+```
+
+### API mocking
+
+#### openapi-backend - <https://www.npmjs.com/package/openapi-backend>
+
```

Comment:
@stoplightio/prism is my go-to for API mocking. It makes more in-depth use of the formats/types/constraints in it's mocking and requires no coding. For example in other tools when I specify a field like

```
myField:
    type: integer
    format: int64
    minimum: 1000
    maximum: 9999
```

Stoplight's `prism` will actually mock values which comply with the constraints of `minimum` and `maximum`.

Reply:
Great suggestion! Thank you

I would replace current suggestion with prism. We had explored both out of the box mocks and mocks that require handlers and found handler ones to be more flexible and working with every api without modification.

Prism is also more mature